src/api/user/src/services/firestore.js

```diff
@@ -1,23 +1,21 @@
 const admin = require('firebase-admin');
 const { logger } = require('@senecacdot/satellite');
 
-process.env.FIRESTORE_EMULATOR_HOST = 'localhost:8088';
-
 // If a private key exists in env.development, run in online mode
 // else run in emulator mode
 if (process.env.PRIVATE_KEY) {
```

Comment:
This isn't defined in any env we have.  Should it be?  Or should you be checking for the presence of all the variables you're about to use below instead?

Reply:
>Or should you be checking for the presence of all the variables you're about to use below instead?

Ideally, yes. Do you know of a succinct way to do that? I was just checking for PRIVATE_KEY because it's the most vital variable required to post to an online firebase db.

PS this should be `process.env.FIREBASE_PRIVATE_KEY` which is an oversight on my end.

Reply:
Here is how I handle a similar thing for auth and SSO details:

https://github.com/Seneca-CDOT/telescope/blob/master/src/api/auth/src/authentication.js#L13-L38

I would do something like this: function that pulls all the expected values off of `process.env` and complains if they are missing, since you can't work without all of them.

Reply:
```
const {
  FIREBASE_TYPE,
  FIREBASE_PROJECT_ID,
  FIREBASE_PRIVATE_KEY_ID,
  FIREBASE_PRIVATE_KEY,
  FIREBASE_CLIENT_EMAIL,
  FIREBASE_CLIENT_ID,
  FIREBASE_AUTH_URI,
  FIREBASE_TOKEN_URI,
  FIREBASE_AUTH_PROVIDER_X509_CERT_URL,
  FIREBASE_CLIENT_X509_CERT_URL,
} = process.env;

// If a private key exists in env.development, run in online mode
// else run in emulator mode
if (
 (FIREBASE_PRIVATE_KEY,
  FIREBASE_PROJECT_ID,
  FIREBASE_PRIVATE_KEY_ID,
  FIREBASE_PRIVATE_KEY,
  FIREBASE_CLIENT_EMAIL,
  FIREBASE_CLIENT_ID,
  FIREBASE_AUTH_URI,
  FIREBASE_TOKEN_URI,
  FIREBASE_AUTH_PROVIDER_X509_CERT_URL,
  FIREBASE_CLIENT_X509_CERT_URL)
) {
  admin.initializeApp({
    credential: admin.credential.cert({
      type: FIREBASE_TYPE,
      project_id: FIREBASE_PROJECT_ID,
      private_key_id: FIREBASE_PRIVATE_KEY_ID,
      private_key: FIREBASE_PRIVATE_KEY,
      client_email: FIREBASE_CLIENT_EMAIL,
      client_id: FIREBASE_CLIENT_ID,
      auth_uri: FIREBASE_AUTH_URI,
      token_uri: FIREBASE_TOKEN_URI,
      auth_provider_x509_cert_url: FIREBASE_AUTH_PROVIDER_X509_CERT_URL,
      client_x509_cert_url: FIREBASE_CLIENT_X509_CERT_URL,
    }),
  });
  logger.debug('Server running in online mode');
} else {
...
}
```

Like this?